python_bindings/src/PyDerivative.cpp

```diff
@@ -0,0 +1,21 @@
+#include "PyDerivative.h"
+
+namespace Halide {
+namespace PythonBindings {
+
+void define_derivative(py::module &m) {
+    auto derivative_class = py::class_<Derivative>(m, "Derivative")
+        .def("__getitem__", [](const Derivative &d, const Func &func) {return d(func);})
+        .def("__getitem__", [](const Derivative &d, const Func &func, int update_id) {return d(func, update_id);})
+        .def("__getitem__", [](const Derivative &d, const Buffer<> &buffer) {return d(buffer);});
```

Comment:
Even if there aren't default-values for args, you should still specify `py::arg("buffer")` here because it allows using named-argument idiom in Python.

Also, nit: spaces around braces, so `{ return d(buffer); });`